src/App.js

```diff
@@ -3,22 +3,28 @@ import { useEffect } from 'react';
 import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
-function useSemiPersistentState() {
-  const [todoList, setTodoList] = useState(() => 
-    JSON.parse(localStorage.getItem('savedTodoList')) || []
-  );
+function App() {
+  const [todoList, setTodoList] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
 
   useEffect(() => {
-    localStorage.setItem('savedTodoList', JSON.stringify(todoList))
+    new Promise((resolve, reject) => {
+      setTimeout(() => {
+        resolve({ data: { todoList: JSON.parse(localStorage.getItem('savedTodoList')) || [] } }); 
+      }, 2000); 
+    })
+    .then((result) => {
+      setTodoList(result.data.todoList);
+      setIsLoading(false);
+    });
+  }, []); 
 
+  useEffect(() => {
+    if (isLoading === false) {
+      localStorage.setItem('savedTodoList', JSON.stringify(todoList));
+    }
   }, [todoList]);
```

Comment:
You need to add `isLoading` as a dependency bacause we have to include all the variables that could change over time used inside the useEffect.